### PR TITLE
[tem-1644] rename stack create

### PR DESCRIPTION
**Pull request by mkrisher**
This PR renames `stack create` to `instance create`. It no longer builds a separate container. This is being reworked so multiple instances of the same stack type can be run in parallel. That work comes in the next PR.

The `install` command no longer made sense, so it has been removed. Users will soon run `init` which will pull the image, and create instances. Instances will have distinct configurations for things like port number, so multiple instances of the same stack can be started and run at the same time.

This PR is just renaming the command and certain references in the configuration file. It draws a better distinction between a Tembo stack definition and an instance as well. Some new objects were identified and broken out.

### CODE REVIEW

File: src/cli/config.rs [hunk 1]
```diff
@@ -136,7 +136,7 @@ impl Config {
 mod tests {
     use super::*;
     use crate::cli::config::Config;
-    use crate::cli::instance::{EnabledExtensions, InstalledExtensions, Instance};
+    use crate::cli::instance::{EnabledExtension, InstalledExtension, Instance};
```
Comment (mkrisher) [inline src/cli/config.rs:139]: annotation: an instance has installed and enabled extensions, a stack has trunk installs and extension.

File: src/cli/stack_error.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+use std::error::Error;
```
Comment (mkrisher) [inline src/cli/stack_error.rs:1]: annotation: just abstracted from the stack file, no changes

File: src/cli/stacks.rs [hunk 3]
```diff
@@ -1,33 +1,42 @@
+// Stacks are defined templates provided by Tembo containing attributes and extensions
```
Comment (mkrisher) [inline src/cli/stacks.rs:1]: annotation: stacks are defined by Tembo, the properties of this object reflect what is provided in the stacks.yml file

File: src/cmd/instance.rs [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+    use crate::cli::instance::{EnabledExtension, InstalledExtension, Instance};
+    use crate::cli::stack_error::StackError;
+    use crate::cli::stacks;
+    use crate::cli::stacks::Stacks;
+    use chrono::prelude::*;
+    use clap::{Arg, ArgAction, ArgMatches, Command};
+    use std::error::Error;
+
+    // example usage: tembo instance create -t oltp -n my_app_db -p 5432
+    pub fn make_subcommand() -> Command {
```
Comment (mkrisher) [inline src/cmd/instance.rs:13]: annotation: this is now `instance create` vs `stack create` and includes three attributes that will allow us to run multiple instances in parallel, including name and port number

File: src/cmd/instance.rs [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+                    .short('p')
+                    .long("port")
+                    .action(ArgAction::Set)
+                    .required(false)
+                    .default_value("5432")
+                    .help("The port number you want to use for this instance (default is 5432)"),
+            )
+    }
+
+    pub fn execute(args: &ArgMatches) -> Result<(), Box<dyn Error>> {
```
Comment (mkrisher) [inline src/cmd/instance.rs:44]: annotation: this no longer builds an container like it did previously, it only stores the configuration, an `instance start` command will use the configuration to run a container given some properties like port number, the configuration will also be used when the `deploy` command is introduced

File: src/cmd/instance.rs [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+    fn persist_instance_config(matches: &ArgMatches) -> Result<(), Box<dyn Error>> {
+        let path = Config::full_path(matches);
+        let mut config = Config::new(matches, &path); // calls init and writes the file
+
+        let r#type = matches.get_one::<String>("type").unwrap();
+        let name = matches.get_one::<String>("name").unwrap();
+        let port = matches.get_one::<String>("port").unwrap();
+
+        let mut instance = Instance {
```
Comment (mkrisher) [inline src/cmd/instance.rs:94]: annotation: this is what gets persisted for each instance created

File: src/cmd/mod.rs [hunk 7]
```diff
@@ -1,3 +1,2 @@
 pub mod init;
-pub mod install;
-pub mod stack;
```
Comment (mkrisher) [inline src/cmd/mod.rs:3]: annoation: bye stack command

File: src/main.rs [hunk 8]
```diff
@@ -22,8 +22,7 @@ fn main() {
 
     let res = match matches.subcommand() {
         Some(("init", sub_matches)) => cmd::init::execute(sub_matches),
-        Some(("install", sub_matches)) => cmd::stack::create::execute(sub_matches),
```
Comment (mkrisher) [inline src/main.rs:25]: annotation: good bye install command

File: tembo/stacks.yaml [hunk 9]
```diff
@@ -1,7 +1,7 @@
 stacks:
   - name: Standard
     description: A balanced Postgres instance optimized for OLTP workloads.
-    stack_version: 0.1.0
+    version: 0.1.0
```
Comment (mkrisher) [inline tembo/stacks.yaml:4]: annotation: removing some redundancy here, small change

**Review by ChuckHend — APPROVED** [review]

**Pull request merged**